data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaCustomConsumer.java

```diff
@@ -65,7 +68,11 @@ public class KafkaCustomConsumer implements Runnable, ConsumerRebalanceListener
 
     private static final Logger LOG = LoggerFactory.getLogger(KafkaCustomConsumer.class);
     private static final Long COMMIT_OFFSET_INTERVAL_MS = 300000L;
+    private static final long IS_EMPTY_CHECK_INTERVAL_MS = 60000L;
     private static final int DEFAULT_NUMBER_OF_RECORDS_TO_ACCUMULATE = 1;
+    static final ConcurrentHashMap<TopicPartition, Boolean> TOPIC_PARTITION_TO_IS_EMPTY = new ConcurrentHashMap<>();
```

Comment:
These static variables will be shared across all buffers and sources, not just the consumers within a single buffer or source. 

It might be more appropriate to create a new class to encapsulate data that needs to be shared across consumers in a single buffer/source. Then create one instance per buffer/source and share to all when creating the consumers.

Reply:
That's a good idea, I can incorporate that